x/stakeibc/types/message_liquid_stake.go

```diff
@@ -32,6 +32,14 @@ func StAssetDenomFromHostZoneDenom(hostZoneDenom string) string {
 	return "st" + hostZoneDenom
 }
 
+func IsStAssetDenom(stAssetDenom string) bool {
```

Comment:
Afaiu this check could fail with assets like `stride` where the prefix is `st`. If we ever have a hz with denom `stx` I think the current RedeemStake flow would allow those holding native `stx` tokens to redeem them for tokens `x` because `IsStAssetDenom` would pass, then `stx` would get converted to `x` by `HostZoneDenomFromStAssetDenom`. There might be logic in the middleware that would catch this and prevent this issue, but we should probably just fix it here at `IsStAssetDenom` by iterating the existing stTokens rather than checking that the first two chars in the string are "st".

Reply:
We actually already have a `CheckIsStToken` that does this exactly. Going to replace `IsStAssetDenom` with that